### PR TITLE
New System Defintions

### PR DESCRIPTION
**Pull request by HexaField**
## Summary

_A summary of changes being made in this PR_

## References

closes https://github.com/EtherealEngine/etherealengine/issues/7363

## Checklist
- [ ] If this PR is still a WIP, convert to a draft
- [ ] When this PR is ready, mark it as "Ready for review"
- [ ] [ensure all checks pass](https://github.com/etherealengine/etherealengine/wiki/Testing-&-Contributing)
- [ ] Changes have been manually QA'd
- [ ] Changes reviewed by at least 2 approved reviewer

## QA Steps

_List any additional steps required to QA the changes of this PR, as well as any supplemental images or videos._

### CODE REVIEW

File: packages/client-core/i18n/en/common.json [hunk 1]
```diff
@@ -51,6 +51,10 @@
     "stats": "Stats",
     "debugOptions": "Debug Options",
     "debug": "Debug Helpers",
+    "preSystems": "Pre Systems",
+    "simulation": "Simulation Group",
+    "subSystems": "Sub System",
```
Comment (speigg) [inline packages/client-core/i18n/en/common.json:56]: Sub systems?

File: packages/client-core/i18n/en/editor.json [hunk 2]
```diff
@@ -602,7 +602,8 @@
     "systemnode": {
       "description": "Attach script.",
       "lbl-filePath": "Url",
-      "lbl-systemUpdateType": "Injection Point",
+      "lbl-insertUUID": "Insertion Point",
```
Comment (speigg) [inline packages/client-core/i18n/en/editor.json:605]: Reference system?

File: packages/client-core/src/recording/RecordingService.ts [hunk 3]
```diff
@@ -56,40 +59,47 @@ export const RecordingFunctions = {
...
+  for (const action of playbackChangedQueue()) {
+    recordingState.playback.set(action.playing ? action.recordingID : null)
   }
+}
 
-  return { execute, cleanup }
+const reactor = () => {
+  useEffect(() => {
+    return () => {
+      removeActionQueue(startRecordingQueue)
```
Comment (speigg) [inline packages/client-core/src/recording/RecordingService.ts:92]: These are no longer necessary in this context

File: packages/client-core/src/social/services/PartyService.ts [hunk 4]
```diff
@@ -455,3 +407,60 @@ export class PartyActions {
...
+  for (const action of patchedPartyUserQueue()) PartyServiceReceptors.patchedPartyUserReceptor(action)
+  for (const action of removedPartyUserQueue()) PartyServiceReceptors.removedPartyUserReceptor(action)
+  for (const action of changedPartyActionQueue()) PartyServiceReceptors.changedPartyReceptor(action)
+  for (const action of resetUpdateNeededActionQueue()) PartyServiceReceptors.resetUpdateNeededReceptor(action)
+}
+
+const reactor = () => {
+  useEffect(() => {
+    return () => {
+      removeActionQueue(loadedPartyQueue)
```
Comment (speigg) [inline packages/client-core/src/social/services/PartyService.ts:436]: These are no longer needed

**Comment by AidanCaruso** [comment]
Just tested this out with bubble pop. Switching the systems in bubble pop to the new definitions and visiting the location url yields these errors that prevent the location from loading:
image

I've run dev-reinit and made sure my modules are up to date.

File: packages/client-core/src/recording/RecordingService.ts [hunk 5]
```diff
@@ -56,40 +59,47 @@ export const RecordingFunctions = {
...
 }
 
-export async function RecordingStateReceptorSystem() {
-  const recordingState = getMutableState(RecordingState)
-
-  const startRecordingQueue = createActionQueue(ECSRecordingActions.startRecording.matches)
-  const recordingStartedQueue = createActionQueue(ECSRecordingActions.recordingStarted.matches)
-  const stopRecordingQueue = createActionQueue(ECSRecordingActions.stopRecording.matches)
-  const playbackChangedQueue = createActionQueue(ECSRecordingActions.playbackChanged.matches)
+const startRecordingQueue = createActionQueue(ECSRecordingActions.startRecording.matches)
```
Comment (speigg) [inline packages/client-core/src/recording/RecordingService.ts:62]: Should be defineActionQueue

File: packages/client-core/src/social/services/PartyService.ts [hunk 6]
```diff
@@ -455,3 +407,60 @@ export class PartyActions {
     type: 'ee.client.Party.RESET_UPDATE_NEEDED' as const
   })
 }
+
+const loadedPartyQueue = createActionQueue(PartyActions.loadedPartyAction.matches)
```
Comment (speigg) [inline packages/client-core/src/social/services/PartyService.ts:411]: defineActionQueue

File: packages/client-core/src/systems/AdminSystem.tsx [hunk 7]
```diff
@@ -41,268 +44,271 @@ import {
...
-    for (const action of inviteCreatedQueue()) AdminInviteReceptors.inviteCreatedReceptor(action)
-    for (const action of invitePatchedQueue()) AdminInviteReceptors.invitePatchedReceptor(action)
-    for (const action of inviteRemovedQueue()) AdminInviteReceptors.inviteRemovedReceptor(action)
-    // for (const action of authSettingRetrievedQueue()) //   AuthSettingsReceptors.authSettingRetrievedReceptor(action)/ }
-    // for (const action of authSettingPatchedQueue()) //   AuthSettingsReceptors.authSettingPatchedReceptor(action)/ }
-    // for (const action of fetchedClientQueue()) //   ClientSettingReceptors.fetchedClientReceptor(action)/ }
-    // for (const action of clientSettingPatchedQueue()) //   ClientSettingReceptors.clientSettingPatchedReceptor(action)/ }
-    for (const action of buildStatusRetrievedQueue()) AdminBuildStatusReceptors.fetchBuildStatusReceptor(action)
-  }
+const fetchedTaskServersQueue = createActionQueue(AdminTaskServerSettingActions.fetchedTaskServers.matches)
```
Comment (speigg) [inline packages/client-core/src/systems/AdminSystem.tsx:47]: defineActionQueue

File: packages/client-core/src/systems/AdminSystem.tsx [hunk 8]
```diff
@@ -41,268 +44,271 @@ import {
...
+  // for (const action of fetchedClientQueue()) //   ClientSettingReceptors.fetchedClientReceptor(action)/ }
+  // for (const action of clientSettingPatchedQueue()) //   ClientSettingReceptors.clientSettingPatchedReceptor(action)/ }
+  for (const action of buildStatusRetrievedQueue()) AdminBuildStatusReceptors.fetchBuildStatusReceptor(action)
+}
 
-  return { execute, cleanup }
+const reactor = () => {
+  useEffect(() => {
+    return () => {
+      removeActionQueue(fetchedTaskServersQueue)
```
Comment (speigg) [inline packages/client-core/src/systems/AdminSystem.tsx:228]: Not needed

File: packages/client-core/src/systems/LoadingUISystem.tsx [hunk 9]
```diff
@@ -36,127 +40,160 @@ import type { WebLayer3D } from '@etherealengine/xrui'
...
+    if (opacity < 0.001) setVisibleComponent(ui.entity, false)
+  })
+}
+
+const reactor = () => {
+  useEffect(() => {
+    return () => {
+      const { ui, mesh } = getState(LoadingUISystemState)
+
+      removeActionQueue(avatarModelChangedQueue)
```
Comment (speigg) [inline packages/client-core/src/systems/LoadingUISystem.tsx:174]: Not needed

File: packages/client-core/src/systems/ProjectUpdateSystem.tsx [hunk 10]
```diff
@@ -1,26 +1,36 @@
...
+  for (const action of clearProjectUpdateQueue()) ProjectUpdateReceptors.clearProjectUpdateReceptor(action)
+  for (const action of setProjectUpdateFieldQueue()) ProjectUpdateReceptors.setProjectUpdateFieldReceptor(action)
+  for (const action of mergeProjectUpdateFieldQueue()) ProjectUpdateReceptors.mergeProjectUpdateFieldReceptor(action)
+}
 
-  return { execute, cleanup }
+const reactor = () => {
+  useEffect(() => {
+    return () => {
+      removeActionQueue(initializeProjectUpdateQueue)
```
Comment (speigg) [inline packages/client-core/src/systems/ProjectUpdateSystem.tsx:23]: Not needed

File: packages/client-core/src/world/ClientSystems.ts [hunk 11]
```diff
@@ -0,0 +1,82 @@
...
+import { WebGLRendererSystem } from '@etherealengine/engine/src/renderer/WebGLRendererSystem'
+import { SceneSystemLoadGroup, SceneSystemUpdateGroup } from '@etherealengine/engine/src/scene/SceneClientModule'
+import { SceneObjectSystem } from '@etherealengine/engine/src/scene/systems/SceneObjectSystem'
+import { ReferenceSpaceTransformSystem } from '@etherealengine/engine/src/transform/systems/ReferenceSpaceTransformSystem'
+import { TransformSystem } from '@etherealengine/engine/src/transform/systems/TransformSystem'
+import { XRAnchorSystem } from '@etherealengine/engine/src/xr/XRAnchorSystem'
+import { XRSystem } from '@etherealengine/engine/src/xr/XRSystem'
+import { XRUISystem } from '@etherealengine/engine/src/xrui/systems/XRUISystem'
+
+export const ClientSystems = () => {
```
Comment (speigg) [inline packages/client-core/src/world/ClientSystems.ts:37]: startClientSystems()?

File: packages/client-core/src/world/DefaultLocationSystems.ts [hunk 12]
```diff
@@ -1,48 +1,28 @@
...
+import { WebcamInputSystem } from '../media/webcam/WebcamInput'
+import { ClientNetworkingSystem } from '../networking/ClientNetworkingSystem'
+import { AvatarUISystem } from '../systems/AvatarUISystem'
+import { LoadingUISystem } from '../systems/LoadingUISystem'
+import { WarningUISystem } from '../systems/WarningUISystem'
+import { WidgetUISystem } from '../systems/WidgetUISystem'
+import { UpdateNearbyUsersSystem } from '../transports/UpdateNearbyUsersSystem'
+import { UserUISystem } from '../user/UserUISystem'
+
+export const useDefaultLocationSystems = (online: boolean) => {
```
Comment (speigg) [inline packages/client-core/src/world/DefaultLocationSystems.ts:16]: Why is this one a hook versus ClientSystems()? Does it make sense to standardize on a single convention for this?

File: packages/client/src/pages/capture/capture.tsx [hunk 13]
```diff
@@ -3,33 +3,32 @@ import { useParams } from 'react-router-dom'
...
 import CaptureUI from '@etherealengine/ui/src/components/tailwind/Capture'
 
-const networkingSystems = {
-  uuid: 'ee.client.core.ClientNetworkingSystem',
-  type: SystemUpdateType.POST_RENDER,
-  systemLoader: () => Promise.resolve({ default: ClientNetworkingSystem })
-} as SystemModuleType<any>
-
-const systems = [...MediaModule(), ...MotionCaptureModule(), networkingSystems]
+const injectCaptureSystems = () => {
```
Comment (speigg) [inline packages/client/src/pages/capture/capture.tsx:20]: startCaptureSystems()?

**Review by speigg — CHANGES_REQUESTED** [review]

File: packages/client-core/src/world/DefaultLocationSystems.ts (hunk 12 above)
Comment (HexaField) [reply packages/client-core/src/world/DefaultLocationSystems.ts:16]: startClientSystems runs amongst other logic, this is standalone

**Pull request merged**